src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -124,10 +124,7 @@ private long GetCountOfAttachmentsForHash(DocumentsOperationContext context, Sli
             Stream stream)
         {
             if (context.Transaction == null)
-            {
-                DocumentsStorage.ThrowRequiresTransaction();
-                return default(AttachmentResult); // never reached
-            }
+                throw DocumentPutAction.ThrowRequiresTransaction();
```

Comment:
OK. I didn't realized that we wrap the throw because of perf, I thought it was purely for readability.
Will revert this specific change.